docs/openapi_spec.json

```diff
@@ -0,0 +1,212 @@
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/v1/teams": {
+      "post": {
+        "tags": ["Teams"],
+        "summary": "Send the names of the iPraktikum teams",
+        "description": "Send a list of all teams/projects that students can work on during this semesters iteration of the iPraktikum course at TUM",
+        "responses": {
+          "200": {
+            "description": "An array of team (names) in JSON format",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                    "items": {
+                      "oneOf": [{
+                      "$ref": "#/components/schemas/Team"
+                    }]
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/v1/allocation": {
+      "get": {
+        "tags": ["Allocation Result"],
+        "summary": "Retrieve the result of the matching/allocation",
+        "description": "After the allocation is concluded, TEASE can report the results (which team each student was assigned to) without needing to repeat/send back all of the information that was initially sent",
+        "responses": {
+          "200": {
+            "description": "An array of tuples with an identifier for the student and the name of the team that they were allocated to",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                    "items": {
+                      "oneOf": [{
+                      "$ref": "#/components/schemas/StudentInTeam"
+                    }]
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "Student": {
+        "type": "object",
+        "properties": {
+          "firstName": {
+            "type": "string"
+          },
+          "lastName": {
+            "type": "string"
+          },
+          "email": {
+            "type": "string"
+          },
+          "tumId": {
+            "type": "string"
+          },
+          "gender": {
+            "$ref": "#/components/schemas/Gender"
+          },
+          "nationality": {
+            "type": "string"
+          },
+          "major": {
+            "type": "string"
+          },
+          "semester": {
+            "type": "number"
+          },
+
+          "germanLanguageLevel": {
+            "type": "string"
+          },
+          "englishLanguageLevel": {
+            "type": "string"
+          },
+
+          "iosDev": {
+            "type": "string"
+          },
+          "appStoreLink": {
+            "type": "string"
+          },
+          "iOSDevExplained": {
+            "type": "string"
+          },
+
+          "introAssessment": {
+            "type": "string"
+          },
```

Comment:
Please add a schema for the assessment options and use that for all the assessments. Also update the name of the property to be more descriptive: It is not clear if that is the self assessment, tutor assessment or PM assessment.

Reply:
I've set the tutor's assessment for the intro course to also use `SkillLevel`. The example CSV uses values ranging from "Not challenging at all" to "Extremely challenging" for the self assessment, would you be alright with me adding a `SelfAssessmentLevel` enum to the schemata?

I'm open to also using `SkillLevel` or potentially shortening the `SelfAssessmentLevel` to "easy", "medium", "challenging", "extremely difficult" or something similar, I could also imagine that some non-native speakers might misunderstand the "hardly challenging" to maybe mean something along the lines of "hard/challening" instead of "barely challenging" although if the options are ordered then I guess it should be clear.